packages/libs/components/src/map/MapVEuMap.tsx

```diff
@@ -376,27 +399,114 @@ function PerformFlyToMarkers(props: PerformFlyToMarkersProps) {
         zoom: mapEvents.getZoom(),
       });
+
+      // check selected markers are within the map viewport/bounds
+      if (
+        selectedMarkers != null &&
+        selectedMarkers.length > 0 &&
+        setSelectedMarkers != null &&
+        zoomLevelToGeohashLevel != null &&
+        setPrevGeohashLevel != null
+      ) {
+        if (prevGeohashLevel === zoomLevelToGeohashLevel(mapEvents.getZoom())) {
+          setSelectedMarkers(
+            (prevSelectedMarkers: markerDataProp[]): markerDataProp[] => {
+              const visibleMarkers = prevSelectedMarkers
+                .map((selectedMarker: markerDataProp) => {
+                  if (mapEvents.getBounds().contains(selectedMarker.latLng)) {
+                    return selectedMarker;
+                  }
+                })
+                .filter((item) => item != null);
+
+              if (visibleMarkers != null)
+                return visibleMarkers as markerDataProp[];
+              else return [];
+            }
+          );
+        } else {
+          // when geohash level is changed
+          setSelectedMarkers([]);
+        }
+      }
+
+      // update preGeohashLevel. Separate condition is made as this is not related to selectedMarkers
+      if (
+        zoomLevelToGeohashLevel != null &&
+        setPrevGeohashLevel != null &&
+        prevGeohashLevel !== zoomLevelToGeohashLevel(mapEvents.getZoom())
+      ) {
+        setPrevGeohashLevel(zoomLevelToGeohashLevel(mapEvents.getZoom()));
+      }
     },
     moveend: () => {
       onViewportChanged({
         center: [mapEvents.getCenter().lat, mapEvents.getCenter().lng],
         zoom: mapEvents.getZoom(),
       });
+
+      // map panning occurred
+      if (setIsPanning != null)
+        setIsPanning((prevMoveend: boolean) => !prevMoveend);
+
+      // check bounds to update selectedMarkers: excluding invisible markers in the selectedMarkers
+      if (
+        selectedMarkers != null &&
+        selectedMarkers.length > 0 &&
+        setSelectedMarkers != null
+      ) {
+        setSelectedMarkers(
+          (prevSelectedMarkers: markerDataProp[]): markerDataProp[] => {
+            const visibleMarkers = prevSelectedMarkers
+              .map((selectedMarker: markerDataProp) => {
+                if (mapEvents.getBounds().contains(selectedMarker.latLng)) {
+                  return selectedMarker;
+                }
+              })
+              .filter((item) => item != null);
+
+            if (visibleMarkers != null)
+              return visibleMarkers as markerDataProp[];
+            else return [];
+          }
+        );
+      }
     },
     baselayerchange: (e: { name: string }) => {
       onBaseLayerChanged && onBaseLayerChanged(e.name as BaseLayerChoice);
     },
+    // map click event: remove selected highlight markers
+    click: () => {
+      removeClassName('highlight-donutmarker');
+      removeClassName('highlight-chartmarker');
```

Comment:
@bobular Yes it could have been used if it belonged to our case, but unfortunately, we have different case.

First of all, the "&" approach is for SaSS/SCSS grammar, not for pure CSS. In addition, the & approach you showed above is for the case that the HTML/div including chart-marker class is a child of the HTML/div including highlighted-marker, e.g.,

```
<div class="highlighted-marker">
    <div class="chart-marker"></div>
 </div> 
```

But in the leaflet, those two classes reside in the same HTML/div code, e.g.,
```
<div class="highlighted-marker chart-marker"></div>
```

This is reason why I had to separately define highlighted-donutmarker and highlighted-chartmarker CSSs, separately.

Reply:
Oh, didn't know that! Then just do

```
.highlighted-marker {
  ...
}
.highlighted-marker.donut-marker {
   ...
}
.highlighted-marker.chart-marker {
  ...
}
```

Reply:
Hi @bobular 👋  

I have made some updates on this PR after checking/testing your suggestions. Will leave my answer to the corresponding one. 

For this case, it turned out that your approach could not be achieved. It is because there is no common CSS that affects highlight: those background-clip and z-index didn't have little impact. Thus, unfortunately we need to keep the current format.